console/client/src/layout/IDELayout.tsx

```diff
@@ -1,30 +1,68 @@
 import { XMarkIcon } from '@heroicons/react/24/outline'
-import { useContext } from 'react'
+import React from 'react'
 import { ModuleDetails } from '../features/modules/ModuleDetails'
 import { ModulesList } from '../features/modules/ModulesList'
 import { Timeline } from '../features/timeline/Timeline'
 import { VerbTab } from '../features/verbs/VerbTab'
-import { SelectedModuleContext } from '../providers/selected-module-provider'
-import { TabType, TabsContext } from '../providers/tabs-provider'
+import { TabsContext, TabSearchParams, TabType } from '../providers/tabs-provider'
 import { headerColor, headerTextColor, panelColor } from '../utils/style.utils'
 import { SidePanel } from './SidePanel'
+import { RequestModal } from '../features/requests/RequestsModal'
+import { useSearchParams, useNavigate , useLocation } from 'react-router-dom'
+import { Tab } from '@headlessui/react'
 
 const selectedTabStyle = `${headerTextColor} ${headerColor}`
 const unselectedTabStyle = `text-gray-300 bg-slate-100 dark:bg-slate-600`
 
 export function IDELayout() {
-  const { selectedModule } = useContext(SelectedModuleContext)
-  const { tabs,activeTab, setActiveTab, setTabs } = useContext(TabsContext)
-
+  const { tabs,activeTab, setActiveTab, setTabs } = React.useContext(TabsContext)
+  const navigate = useNavigate()
+  const location = useLocation()
+  const [ searchParams ] = useSearchParams()
+  
+ 
```

Comment:
Can the linter catch stuff like this? I think prettier might have before, but it seems like the `lint:fix` would remove extra whitespace like this.

Reply:
No rule for that but I also don't particularly concern myself with line space except for at the end of the file. I'll look for a rule that does that if it matters to you.

Reply:
I mostly just want the linter to keep all our code consistent. For example if a developer adds like 8 lines of whitespace that probably isn't what everybody wants so the linter should help us out automatically.

The autoformatting seems fine with stuff like this, which seems off to me:
![Screenshot 2023-08-24 at 10 42 35 AM](https://github.com/TBD54566975/ftl/assets/51647/721a4ddf-4b77-4f98-b100-2279cb0bde43)

Reply:
https://github.com/TBD54566975/ftl/pull/291